src/components/enterprise-user-subsidy/UserSubsidy.jsx

```diff
@@ -10,32 +10,70 @@ import { LoadingSpinner } from '../loading-spinner';
 import {
   useCatalogData, useLearningPathData,
 } from './data/hooks';
-import { LOADING_SCREEN_READER_TEXT } from './data/constants';
+import {
+  LOADING_SCREEN_READER_TEXT, filterInitial, filterOptions, filterOptionsExpanded,
+} from './data/constants';
 
 export const UserSubsidyContext = createContext();
 
 const UserSubsidy = ({ children }) => {
   const { enterpriseConfig } = useContext(AppContext);
-  const [catalogData, isLoadingCatalogData] = useCatalogData(enterpriseConfig.uuid);
+  const [catalogFilter, setCatalogFilter] = useState(filterInitial);
+  const [catalogData, isLoadingCatalogData] = useCatalogData({
+    enterpriseId: enterpriseConfig.uuid,
+    filter: catalogFilter,
+  });
   const [learningPathData, isLoadingLearningPathdata] = useLearningPathData();
 
   const isLoading = isLoadingCatalogData || isLoadingLearningPathdata;
 
+  const toggleFilter = useCallback((group, options) => {
+    setCatalogFilter(currentFilter => {
+      let newFilterValues = [...currentFilter[group]];
+      options.forEach(option => {
+        if (newFilterValues.includes(option)) {
+          newFilterValues = newFilterValues.filter(value => {
+            if (filterOptionsExpanded[group]) {
+              return !filterOptionsExpanded[group][option].includes(value);
+            }
+            return value !== option;
+          });
+        } else {
+          newFilterValues = newFilterValues.concat(
+            filterOptionsExpanded[group] ? filterOptionsExpanded[group][options] : option,
+          );
+        }
+      });
```

Comment:
```suggestion
      let newFilterValues = [...currentFilter[group]];
      const allOptions = filterOptionsExpanded[group] ? filterOptionsExpanded[group][options] : options;

      allOptions.forEach(option => {
        newFilterValues = newFilterValues.includes(option)
          ? newFilterValues.filter(value => value !== option) : [...newFilterValues, option];
      });
```
Multiple conditions in callback functions make it harder to grasp what you want to achieve. It looks more complicated than it is. I would refactor it as shown in the suggestion.
The whole `filterOptionsExpaned` serve to extend `options` from one to multiple `ids` so I would move this logic to the `allOptions` assignment.

And I try to avoid multiple assignments to the same variable if it can be simply solved with 1-level ternary. Otherwise `if` would be prefered.

Reply:
Thanks for the comment. I tend to be more verbose (thus the if/else branching) where the intent of the code might not be obvious at first glance (which it isn't with the ternarys and a lot logic). Will update with the code you shared.